ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -115,6 +126,38 @@
     this.waitEnabled = !replicatingFromRemoteColo;
     this.replicatingOverSsl = replicatingOverSsl;
     this.datacenterName = datacenterName;
+    for (Map.Entry<DataNodeId, List<RemoteReplicaInfo>> entry : replicasToReplicateGroupedByNode.entrySet()) {
+      for (RemoteReplicaInfo info : entry.getValue()) {
+        replicatedPartitions.add(info.getReplicaId().getPartitionId());
+      }
+    }
+  }
+
+  /**
+   * Enables/disables replication on the given {@code id}.
+   * @param ids the {@link PartitionId}s to enable/disable it on.
+   * @param enable whether to enable ({@code true}) or disable.
+   */
+  void controlReplicationForPartitions(List<? extends PartitionId> ids, boolean enable) {
+    lock.lock();
+    try {
+      for (PartitionId id : ids) {
+        if (replicatedPartitions.contains(id)) {
+          if (enable) {
+            replicationDisabledPartitions.remove(id);
```

Comment:
My bad, those are different maps - added this check, there is no correctness issue in the current code but this suggestion avoids unnecessary signaling.